cmake/StandardProjectSettings.cmake

```diff
@@ -0,0 +1,370 @@
+include(GNUInstallDirs) # Standard install dirs
+
+# Uniform how we define BUILD_STATIC or BUILD_SHARED_LIBS (common practice)
+if(DEFINED BUILD_STATIC)
+    if(DEFINED BUILD_SHARED_LIBS)
+        if(${BUILD_SHARED_LIBS} AND ${BUILD_STATIC})
+            message(FATAL_ERROR "Conflicting arguments for BUILD_SHARED_LIBS=${BUILD_SHARED_LIBS} and BUILD_STATIC=${BUILD_STATIC}")
+        endif()
+    else()
+        set(BUILD_SHARED_LIBS NOT ${BUILD_STATIC})
+    endif()
+else()
+    if(NOT DEFINED BUILD_SHARED_LIBS)
+        set(BUILD_SHARED_LIBS ON)
```

Comment:
Maybe it's simpler to use an `option` for this?

```
option(BUILD_SHARED_LIBS "Build dynamically linked library instead of a static library." ON)
```

This allows us to give a description of the option, and to provide a default value. It'll get cached. You'd still need to do the handling of the other option to see whether they conflict, but it may be easier this way.

Reply:
I will remove the build static flag and use the cmake global variable BUILD_SHARED_LIBS like is done in libArcus